src/WebJobs.Extensions.DocumentDB/Trigger/DocumentDBTriggerAttribute.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Extensions
+{
+    using System;
+    using Microsoft.Azure.WebJobs.Extensions.DocumentDB;
+
+    /// <summary>
+    /// Defines the [DocumentDBTrigger] attribute
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1019:DefineAccessorsForAttributeArguments")]
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public sealed class DocumentDBTriggerAttribute : Attribute
```

Comment:
Make the `ConnectionString`, `LeaseConnectionString`, etc, have public setters, then we'll use that. An example from WebJobs here: https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/ExtensionsSample/Samples/DocumentDBSamples.cs#L52 (note that `Id` isn't a part of the constructor).

In Functions, we'll have to write some 'glue' code that creates the attribute based on the function.json; we'll use the properties there as well.